### ISSUE TITLE
teams crashing with regiontoshare

### ISSUE THREAD

**Issue by fernando-vs** [tom-englert/RegionToShare#71]
**Environment**
- Operating System: (Win10/Win11)

windows 11

- Collaboration Tool: (Teams/Webex/...)

teams

**Describe the bug**
I have an ultrawide monitor, and i use teams at work with regiontoshare, unfortunately since yesterdey ive been having issues with the app. the call in teams crashes after i start sharing the app, not even restart would bring it back. i use regiontoshare with a resolution of 2821x1410 at least and 15FPS. the person in the other side mentions that he sees a black screen or the screen is frozen. 

**To Reproduce**
as mentioned, i share the app with teams in an ultrawide monitor (5120x1440), the app is set according to the following: 2821x1410 at least and 15FPS. 

**Expected behavior**
before yesteday, the app was working just fine i was able to share the monitor for a whole meeting.

**Comment by tom-englert** [comment]
> the call in teams crashes 

So this is an issue in teams.

 I also experienced this sometimes, but don't know how to reproducible report this to Microsoft.

**Comment by mjburley** [comment]
I have been having the same issue. An update in Teams broke how Teams works with Region To Share. Mind you, most updates in Teams breaks something, and it seems to update every other day now. 

This was the best screen sharing app I have used on a widescreen monitor. Now I use Picture in Picture to achieve the same, which is far from perfect, or Power Toys has something similar but not as good as this.

**Comment by an0wak** [comment]
I have similar issue with 49" screen. Using FancyZones to break up the screen into segments. Snap RemoteToShare into one of my segments.

It has the same behaviour in Teams and WebEx. As soon as I move my mouse in the Region area it crashes. So remote viewer initially sees the session and then it freezes on them. I lose the "red" share border.

OS: Windows 10
Teams Version: 1.7.00.6058 (64-bit). It was last updated on 11/03/2024.
WebEx Version: 43.2.0.25211

Tested it with various resolutions. Stuck with 1920x1080 15FPS for testing - doesn't seem like a resolution issue.

**Comment by rpeveng** [comment]
Same here, using a 5120*1440 pixels screen.

@mjburley What "Picture in Picture" exactly do you use now?

**Comment by tom-englert** [comment]
@an0wak @rpeveng it's of no use reporting here that Teams has bugs, you need to report this to Microsoft

**Comment by TylerMitton** [comment]
Is this still an issue for others? I'm having issues with regiontoshare and teams, but want to make sure it is the same teams issue. thanks!

**Comment by pmarmol86** [comment]
Yes I am having the same issue and as @an0wak described is happening on both teams and webex. Not sure if this is a microsoft bug since it is happening on webex as well.
What I've done to temporary solve this in teams is cleared the Teams cache folder in these two locations:
_**C:\Users\YOURUSER\AppData\Local\Packages\MSTeams_8wekyb3d8bbwe
C:\Users\YOURUSER\AppData\Roaming\Microsoft\Teams**_
but this means I have to reconfigure the app everytime I do that.
@tom-englert maybe a temp or persistency file for regiontoshare is allocated in the teams temp folder?

**Comment by tom-englert** [comment]
@pmarmol86 RTS doesn't know about Teams and doesn't interact with it.

**Comment by ngbrown** [comment]
Recently, when trying to share the RegionToShare window from Teams, it is causing Teams to malfunction. The screen stops updating for the other user, and then when leaving the meeting, the meeting window won't close. I have to exit completely out of Teams and re-launch it before I can share any other windows.

**Comment by tom-englert** [comment]
@ngbrown this is the issue tracker for RTS, issue of Teams must be reported to Microsoft

**Comment by radhaksri** [comment]
I've been seeing the same issue time and again. The screen stops updating for other users, and teams just hangs - I have to kill it through the task manager before relaunching it. 

There is some aspect of RTS that is making Teams crash out. Not sure where the issue is - Teams or RTS - but we definitely need help here. 

I switched to Ultra Wide Screen Share 2.0 - seems going fine so far. Will watch out to see how it goes.

**Comment by kmtchl** [comment]
Seeing exactly the same issue as above for months. It's a shame that any issue raised like this is being shut down and suggested it needs raising with Microsoft. Appreciate its unlikely a RTS issue but any advice for users at all would be great. 

Unfortunately it makes this great little application unusable. Will try Ultra Wide Screen Share 2.0 as @radhaksri suggested

**Comment by tom-englert** [comment]
@kmtchl if you have the slightest idea how to fix a bug in teams from here, let me know and I'll do it.

**Comment by kmtchl** [comment]
@tom-englert apologies if that came across negatively. But the first line of the readme indicates it works with Teams but it hasn't done for a long time for many users.

Have you experienced the issues we're talking about yourself?

**Comment by tom-englert** [comment]
If I start sharing from Teams when RTS is in active mode, it works most of the time.

**Comment by deckn09** [comment]
> If I start sharing from Teams when RTS is in active mode, it works most of the time.

I always do it like this (RTS in active mode, then start sharing), but still encounter the problem. Also, it doesn't occur once in a while, but nearly every time, so that I stopped using RTS (because killing + restarting Teams and then rejoining the meeting is a pain in the a**)

**Comment by OmiCron07** [comment]
> > If I start sharing from Teams when RTS is in active mode, it works most of the time.
> 
> I always do it like this (RTS in active mode, then start sharing), but still encounter the problem. Also, it doesn't occur once in a while, but nearly every time, so that I stopped using RTS (because killing + restarting Teams and then rejoining the meeting is a pain in the a**)

I feel you, I have the same problem, I had to ditched RTS for `Ultra Wide Screen Share 2.0`.

**Comment by XIAIX-code** [comment]
> There's an alternative here that can fix this problem; (with all my respect to RegionToShare and the owner)
> 
> https://github.com/salihcantekin/RustFrame

And with all due respect to YOU, that's kinda shitty to use this thread to promote yourself.  IMO.
